src/deluge/util/functions.cpp

```diff
@@ -1992,30 +1994,63 @@ int32_t getWhichKernel(int32_t phaseIncrement) {
 	}
 }
 
-void dissectIterationDependence(int32_t iterance, int32_t* getDivisor, int32_t* getWhichIterationWithinDivisor) {
-	int32_t value = (iterance & 127) - 1;
-	int32_t whichRepeat;
+// This methods extracts from the iterance parameter (which is an uint16_t) the divisor and the iteration bits
+// The 'divisor' is the total number of iterations
+// The 'iteration bits' is a byte where each of its 8 bits represents a boolean on
+// whether the iteration step is active or not
+void dissectIterationDependence(int32_t iterance, int32_t* getDivisor, int32_t* getWhichIterationBitsWithinDivisor) {
```

Comment:
an iterance class that has seperate fields would be a lot more readable than this. Packing them into one is a questionable design choice and we definitely shouldn't stick with it

For storage or wherever the small representation is needed just add to/from uint32_t methods to the class so all practical use is done with a normal class

Reply:
Good idea to refactor this into a class